### ISSUE TITLE
reorg directory structure

### ISSUE THREAD

**Issue by ry** [denoland/deno#2852]
After https://github.com/denoland/deno/pull/2825 lands, I want to change the layout of our source tree. #2825 adds two new crates, bringing our total to 4. I think we need to move them all into a special src directory to make grepping easier, and easier for newbies to navigate.

After #2825, the //js directory is only used by `cli_snapshots`, I think it should be moved to a subdirectory of that crate.

Thus I propose

```
//src/
//src/core/
//src/core/libdeno/
//src/cli/
//src/cli/ops
//src/cli_snapshots/
//src/cli_snapshots/js/
//src/deno_typescript/
```

deno_typescript is a long name - I'd like to use something shorter. But I'm not sure what a good name would be that is descriptive yet short.. I think we can punt on that.

Also I am strongly considering merging the deno_std tree into this repo. I think `//src/std` would be logical. I will do that in a separate PR, but just want to mention it.

**Comment by bartlomieju** [comment]
> After #2825, the //js directory is only used by cli_snapshots, I think it should be moved to a subdirectory of that crate.

How about splitting that into `src/runtime` and `src/compiler`? Once GN is not needed anymore both of them would contain minimal `build.rs` files (just different entry points).

> deno_typescript is a long name - I'd like to use something shorter. But I'm not sure what a good name would be that is descriptive yet short.. I think we can punt on that.

`ts_bundler`?

> Also I am strongly considering merging the deno_std tree into this repo. I think //src/std would be logical. I will do that in a separate PR, but just want to mention it.

Why not a git submodule? That would still keep PRs and issues separate - should be easier to manage than having everything in this repo.

**Comment by nayeemrmn** [comment]
> Why not a git submodule? That would still keep PRs and issues separate - should be easier to manage than having everything in this repo.

I think that's how it currently is... they're vendored in the deps cache in `//js/deps`.

**Comment by ry** [comment]
> Why not a git submodule? 

So git tags applied to the main deno apply also to std. Also I find it hard to keep track of PRs in the other repo...

**Comment by bartlomieju** [comment]
> > Why not a git submodule?
> 
> So git tags applied to the main deno apply also to std. 

Got it 👍although I think it may obscure build process again

>Also I find it hard to keep track of PRs in the other repo...

Fair enough, I'm just fearing for that once we get more adoption

**Comment by bartlomieju** [comment]
@ry done or do you still want to move everything under `src/`?

**Comment by ry** [comment]
Done. I’m not going to pursue the src dir.
